src/server/pps/server/api_server_test.go

```diff
@@ -459,6 +459,61 @@ func TestCreatePipelineV2(t *testing.T) {
 	require.False(t, req.Autoscaling, "spec must override default")
 }
 
+func TestRerunPipeline(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	env := realenv.NewRealEnv(ctx, t, dockertestenv.NewTestDBConfig(t))
+
+	repo := "input"
+	pipeline := "pipeline"
+	require.NoError(t, env.PachClient.CreateRepo(pfs.DefaultProjectName, repo))
+	var pipelineTemplate = `{
+		"pipeline": {
+			"project": {
+				"name": "{{.ProjectName | js}}"
+			},
+			"name": "{{.PipelineName | js}}"
+		},
+		"transform": {
+			"cmd": ["cp", "r", "/pfs/in", "/pfs/out"]
+		},
+		"input": {
+			"pfs": {
+				"project": "default",
+				"repo": "{{.RepoName | js}}",
+				"glob": "/*",
+				"name": "in"
+			}
+		},
+		"autoscaling": false
+	}`
+	tmpl, err := template.New("pipeline").Parse(pipelineTemplate)
+	require.NoError(t, err, "template must parse")
```

Comment:
I think that some existing tests use this templaing pattern, but I'm not sure it's that good.  Just filling out the RPC you want to run and calling it seems easier to me:

```go
spec := &pps.CreatePipelineRequest{
    Pipeline: &pps.Pipeline{
        Name: pipelineName,
         ...
     }
}
js, err := protojson.Marshal(spec)
...
req := &pps.CreatePipelineV2Request{ CreatePipelineRequestJson: js }
...
```